docs/source/parsing_best_practices.md

```diff
@@ -0,0 +1,240 @@
+# nvstrings Parsing Best Practices
+
+
+## Regex Performence Tips:
+When working with strings on GPUs, performance is heavily dependent on data and the cuStrings functions you use, especially when doing character/substring substitution and removal.
+
+If you need to write on regex for your pre-processing tasks, follow the below guidelines to achieve best performance.
+
+### Sorting by length:
+
+If your data has unequal length distribution (For example, the longest string might be 1000x longer than the shortest string) and you are doing heavy regex heavy operations, sorting the strings can help to improve the performance.
+
+##### Why:
+
+As regexes operate one character at a time, longer length strings take longer to evaluate and operating on strings in parallel improves throughput.   The GPU works most efficiently when adjacent threads in a warp work in lock step. Wildly different string lengths within adjacent threads means that the threads operating on shorter strings become idle, reducing utilization.
+
+Sorting helps to obtain better wrapping helping adjacent threads in a warp work in lock step to improve utilization . 
```

Comment:
Better 'wrapping'?  Assume you mean better 'warping' though I don't think that is real term. How about this:
_Sorting helps to improve performance by helping adjacent threads in a warp work in lock step to improve utilization_

Reply:
Should be fixed after commit: https://github.com/rapidsai/custrings/pull/397/commits/8f5ce58efd95160ba4ff0f2d2014f494da1adb34